core/arch/arm/kernel/elf_load.c

```diff
@@ -357,9 +358,11 @@ static TEE_Result e32_process_rel(struct elf_load_state *state, size_t rel_sidx,
 			return TEE_ERROR_BAD_FORMAT;
 
 		/* Check the address is inside TA memory */
-		if (shdr[sym_tab_idx].sh_addr > state->vasize ||
-		    (shdr[sym_tab_idx].sh_addr +
-				shdr[sym_tab_idx].sh_size) > state->vasize)
+		if (ADD_OVERFLOW(shdr[sym_tab_idx].sh_addr,
+				 shdr[sym_tab_idx].sh_size, &sh_end))
+			return TEE_ERROR_BAD_FORMAT;
+		if (shdr[sym_tab_idx].sh_addr >= state->vasize ||
```

Comment:
Given that `sh_end` is guaranteed to as large or larger than `shdr[sym_tab_idx].sh_addr`, isn't this first part of the `if` redundant?

Reply:
Hum, yeah, that is correct.